New branch "except M1"

Y-Pazniak opened the pull request:
M1 branch works with new branches - need to fill another branches.

Y-Pazniak committed:
Added two empty branches.
M1 branch works with new branches - need to fill another branches.

Y-Pazniak committed:
Added two empty branches.
M1 branch works with new branches - need to fill another branches.

Y-Pazniak committed:
new step for "except_m1" branch has been added

Y-Pazniak committed:
weight selection for "except_m1 -> n1_n3" has been added

Y-Pazniak committed:
some prices for exceptM1 - n1-n3 has been added

Y-Pazniak committed:
Incorrect command ignoring was fixed.

Except m1 -> n1-n3 -> all branch with prices was added

Y-Pazniak committed:
Branch with prices for "except m1 -> m2-m3 -> electric engine" has been added

Y-Pazniak committed:
Branch with prices for "except m1 -> m2-m3 -> electric engine" has been added

Y-Pazniak committed:
Experiment with common engine type has failed.
Command tree json has been shorted using one common command for N1-N3 weights

Y-Pazniak committed:
Experiment with common engine type has failed.
Command tree json has been shorted using one common command for N1-N3 weights

Y-Pazniak committed:
branch "except m1 -> m2 -> engine volume" has been added WITHOUT price calculation

Y-Pazniak committed:
branch "except m1 -> m2 -> engine volume" has been finished

branch "except m1 -> truck units" has been started, types of trucks units has been added

MaximPilipeiko commented on src/main/java/by/custom/utilcalculator/service/UserProgressManager.java:
```diff
@@ -134,26 +169,61 @@ public String processEngineType(final Command requestingCommand, final String ch
         userProgressStorage.save(userProgress);
-        message = messagesCreator.getSummaryAnswer(userProgress);
-        return message;
+        System.out.println("truck units");
```
Is it needed?

Y-Pazniak replied:
No, we don't need sout here. I don't have it in my last version - probably it is an old commit or smth - I don't have any System commands in my code
![image](https://github.com/user-attachments/assets/b8d74246-e072-4026-a381-88d9b53534c9)

Y-Pazniak committed:
branch "except m1 -> truck units" has been finished

Y-Pazniak committed:
branch "except m1" has been completely finished

MaximPilipeiko commented on src/main/resources/tree.json:
```diff
@@ -78,9 +73,101 @@
+          "key": "TRUCK_UNITS",
+          "children": [
+            {
+              "key": "TRUCK_UNITS_EXCEPT_6_CLASS",
```
If I understand correctly what you would like to achieve, It's better to use for this cases `_OTHER`
```suggestion
              "key": "TRUCK_UNITS_OTHER",
```

Y-Pazniak replied:
In this case yes, we can rename - we have only 2 types of trucks: 6 class and not-6-class.
Renamed

MaximPilipeiko commented on src/main/resources/tree.json:
```diff
@@ -78,9 +73,101 @@
     },
     {
       "key": "EXCEPT_M1",
```
```suggestion
      "key": "OTHER",
```

MaximPilipeiko replied:
Let's disscuss it on a call

Y-Pazniak replied:
OK, let's discuss. In my opinion "other" is too wide - we can use it everywhere, but "except m1" is rather narrow and when I see "except m1" I clearly understand, that it is any type of vehicle except passenger auto. Probably, we can use "m1_vehicles" and "other_vehicles"?

MaximPilipeiko replied:
Please, add description

MaximPilipeiko commented on src/main/java/by/custom/utilcalculator/service/UserProgressManager.java:
```diff
@@ -134,26 +169,90 @@ public String processEngineType(final Command requestingCommand, final String ch
+
+    public String processTruckUnitWeight(final Command requestingCommand, final String chatID) throws UtilsborException {
+        final UserProgress userProgress = userProgressStorage.get(chatID);
+        if (!UserProgressValidator.validateCommand(Command.TRUCK_UNIT_WEIGHT, userProgress)) {
```
Is command really should be hardocoded?
```suggestion
        if (!UserProgressValidator.validateCommand(requestingCommand, userProgress)) {
```

Y-Pazniak replied:
In this case yes, thank you for noticing it. 
Sometimes I have a node with different answer. Example: 
![image](https://github.com/user-attachments/assets/eab7f82c-b104-4647-a606-699b8ebe86ee)

But i don't care about user answer - I know that despite user answer he has only one possible next step. 
Example.
![image](https://github.com/user-attachments/assets/0b5fc889-04ba-4055-95c0-ce4089d588f4)
 
Whatever user choose, next step - age auto.
For this cases I decided to unite commands in one common command to avoid useless extra nodes in tree. 
![image](https://github.com/user-attachments/assets/3a928599-c41d-4a0f-af06-54a3d1539203)
This is possibilities for user commands. 

![image](https://github.com/user-attachments/assets/1a9962e4-e71e-4077-b55c-5ed4161c9a91)
This is how it looks in the tree. 

And I mustn't check the particular command, I have to check only does this command belongs to correct group of commands or not and I need to hardcode this command.

MaximPilipeiko commented on src/main/java/by/custom/utilcalculator/service/UserProgressManager.java:
```diff
@@ -134,26 +169,90 @@ public String processEngineType(final Command requestingCommand, final String ch
+
+    public String processTrailersO4Type(final Command requestingCommand, final String chatID) throws UtilsborException {
+        final UserProgress userProgress = userProgressStorage.get(chatID);
+        if (!UserProgressValidator.validateCommand(Command.TRAILERS_O4_TYPE, userProgress)) {
```
Is command really should be hardocoded?
```suggestion
        if (!UserProgressValidator.validateCommand(requestingCommand, userProgress)) {
```

MaximPilipeiko commented on src/main/java/by/custom/utilcalculator/service/CalculatorRouter.java:
Let's disscuss it

MaximPilipeiko reviewed (changes requested):
Looks good. Just need not discuss to understand some things

MaximPilipeiko commented on src/main/java/by/custom/utilcalculator/controller/Bot.java:
```diff
@@ -22,8 +22,9 @@ public static Bot getBot() {
+        //need MessageRouter object to work with it after - to proceed user's messages
         messageRouter = MessageRouter.getInstance();
+        //don't need commandTree object in future - only to load tree.json file to build correct branches of calculator
         CommandTree.getInstance();
```
```suggestion
        CommandTree.init();
```

Y-Pazniak replied:
Fixed.

MaximPilipeiko commented on src/main/java/by/custom/utilcalculator/domain/constants/Command.java:
```diff
@@ -4,67 +4,95 @@ public enum Command {
     private final String command;
 
-    Command (final String command) {
+    Command(final String command) {
```
```suggestion
    Command(final String command, final @Nullable Command reference) {
```

Y-Pazniak replied:
Done.

Y-Pazniak committed:
Comments to the tree.json have been added.
More proper naming for nodes has been added.

Y-Pazniak committed:
Tree init has been removed from Bot.java
I've discovered we don't need it during bot loading - it loads with first "/start" command of any user and then we can use this singleton object anytime

Y-Pazniak committed:
Families for commands have been added

Y-Pazniak committed:
wrong path to properties was fixed

Y-Pazniak committed:
Adding steps into Command in process

Y-Pazniak committed:
Adding steps into Command has been finished

MaximPilipeiko commented on src/main/java/by/custom/utilcalculator/domain/constants/steps/ExceptM1TransportType.java:
```diff
@@ -0,0 +1,5 @@
+package by.custom.utilcalculator.domain.constants.steps;
+
+public enum ExceptM1TransportType implements StepsIndicator {
```
Please, find and change ExceptM1 to smth like BusesAndTrucks
```suggestion
public enum BusesAndTrucksTransportType implements StepsIndicator {
```

Y-Pazniak replied:
Done.

MaximPilipeiko commented on src/main/java/by/custom/utilcalculator/domain/constants/steps/GeneralTransportType.java:
```diff
@@ -0,0 +1,5 @@
+package by.custom.utilcalculator.domain.constants.steps;
+
+public enum GeneralTransportType implements StepsIndicator{
+    M1, EXCEPT_M1, SELF_PROPELLED_VEHICLES
```
```suggestion
    M1, BUSSES_AND_TRUCKS, SELF_PROPELLED_VEHICLES
```

Y-Pazniak replied:
Fixed.

MaximPilipeiko commented on src/main/java/by/custom/utilcalculator/domain/constants/Command.java:
```diff
@@ -1,70 +1,117 @@
+    private final Step nextStep;
 
-    Command (final String command) {
+    Command(final String command, final Command family, final Step step) {
```
Please, rememeber me what we decided to store in `step`. If it's the next step it might be a problem. We may not know which language will be used at this level. The next step is defined in `tree.json`

Y-Pazniak replied:
We decided to store there current step. My personal decision was to store there next step. Possibly, I was wrong.

MaximPilipeiko commented on src/main/java/by/custom/utilcalculator/service/BundleResourcesServant.java:
```diff
@@ -26,7 +19,12 @@ private static class BundleResourcesCreatorHolder {
+        if (bundle.containsKey(stringToGetFromBundle)) {
+            return bundle.getString(stringToGetFromBundle);
+        } else {
+            System.out.println("no such string in bundle, check the path");
```
Which path?
```suggestion
            System.out.printf("no such string in bundle, check the path `%s`\n", stringToGetFromBundle);
```

Y-Pazniak replied:
Fixed.

MaximPilipeiko commented on src/main/java/by/custom/utilcalculator/service/BundleResourcesServant.java:
```diff
@@ -26,7 +19,12 @@ private static class BundleResourcesCreatorHolder {
     }
 
     public String getString(final String stringToGetFromBundle) {
```
I think this name is more attractive and informative enough.
```suggestion
    public String getString(final String localizationKey) {
```

Y-Pazniak replied:
Renamed.

MaximPilipeiko commented on src/main/java/by/custom/utilcalculator/service/UserProgressManager.java:
```diff
@@ -32,25 +32,74 @@ public void createNewUserProgress(final String chatID) throws UtilsborException
+        }
+        userProgress.setNextStep(requestingCommand.getNextStep());
+        switch (requestingCommand) {
+            case LESS_2_TONS -> userProgress.setTransportWeightN1N2N3(N1N3TransportWeight.LESS_2_TONS);
```
You have already create association between Command and StepIndicator in `fillFieldsToCommandsMap`. Please, use this map instead of duplication. It will help to avoid potential inconsistency if smth will be missed.

Y-Pazniak replied:
I like your idea very much. Unfortunately, it seems I have a problem.
We have userProgress object with fields for price calculation. We need to fill fields with 100% accuracy. We need to know which type of vehicle in particular to calculate a correct price. 
But during filling fillFieldsToCommandsMap sometimes we use families instead of particular field. 
For example. For a price calculation I have to know the volume of an engine. But for the fillFieldsToCommandsMap is enough to know only that volume field was filled - no difference, which in particular. 
I see no solutions right now, but we can discuss it and find the best way.

MaximPilipeiko commented on src/main/java/by/custom/utilcalculator/domain/UserProgress.java:
```diff
@@ -8,50 +8,118 @@
+    private TrailerO4Type trailersO4Type = null;
     private final String chatID;
-    private Step currentQuestion;
+    private Step nextStep;
```
Not sure that it's good decision store nextStep instead of current

Y-Pazniak replied:
Highly likely you are right. It was completely my decision, I disobeyed your order to save here current step

MaximPilipeiko commented on src/main/java/by/custom/utilcalculator/domain/UserProgress.java:
```diff
@@ -8,50 +8,118 @@
         return countryOrigin;
     }
 
     public Command[] getUserPath() {
```
Already forgot how we use the user path. Please, remember me to discuss it on our call. Now it looks strange.

MaximPilipeiko replied:
Too many steps. Possible we have to fill it after each user command and not calculate it. We will discuss it.

Y-Pazniak replied:
I use getUserPath().
1. For getting the current user's node. Class CommandTree - method getNode(final UserProgress userProgress) returns current user's node. 
2. For user path string building (a line about previous user's steps).
3. For debugging to check this array filling - what commands there are.

MaximPilipeiko commented on src/main/resources/tree.json:
```diff
@@ -1,28 +1,33 @@
 {
   "key": "START",
```
Could you please, draw a visualization of this tree? Even physical on paper will be enough.

Y-Pazniak replied:
[chatbot-scheme.pdf](https://github.com/user-attachments/files/16522953/chatbot-scheme.pdf)

MaximPilipeiko commented on src/main/java/by/custom/utilcalculator/domain/UserProgress.java:
```diff
@@ -71,75 +137,47 @@ public CarAge getCarAge() {
     }
 
-    private void cleanStepsAfterCurrent(final int stepCleaner) {
+    private void cleanStepsAfterCurrentM1Branch(final int stepCleaner) {
```
It's strange that we have two different methods `cleanStepsAfterCurrentM1Branch` and `cleanStepsAfterCurrentExceptM1Branch` instead of one `cleanStepsAfterCurrent `

Y-Pazniak replied:
Now we have three of them :) 
I can try to rewrite it into one and to check what will happen.

Y-Pazniak committed:
getUserPath array was replaced by list, no empty cell more

Y-Pazniak committed:
array has been replaced by list, rewriting of list has been added

Y-Pazniak committed:
passenger m1 cars useless type of owners were removed

Y-Pazniak committed:
bounding tree.json and step enum is in progress

Y-Pazniak committed:
binding tree.json and step enum for buses and trucks (passenger cars) 50% done

Y-Pazniak committed:
binding tree.json and step enum for buses and trucks (passenger cars) 50% done

Y-Pazniak committed:
binding tree.json and step enum for buses and trucks (passenger cars) has been finished

Y-Pazniak committed:
binding tree.json and step enum for buses and trucks (passenger cars) has been finished

MaximPilipeiko commented on src/main/java/by/custom/utilcalculator/domain/UserProgress.java:
```diff
@@ -5,155 +5,200 @@
+    }
+
+    public void setTrailerO4Type(final TrailerO4Type trailersO4Type) {
+        cleanStepsAfterCurrentM1Branch(3);
```
You have to calculate the step number. I expect to see something like this.
```suggestion
        var newStep = TreeMediator.getNextStep(this);
        cleanStepsAfterCurrent(newStep.number);
        this.trailersO4Type = trailersO4Type;
        updateUserPath(trailersO4Type);
```

Y-Pazniak replied:
In my head we don't need cleanSteps here future - it should works automatically. I'm going to write to you my solution in telegram and I'll be happy to discuss it with you.

MaximPilipeiko commented on src/main/java/by/custom/utilcalculator/domain/UserProgress.java:
```diff
@@ -5,155 +5,200 @@
+    }
+
+    private void addUserStatusToPath(final StepsIndicator stepsIndicator) {
+        final Map<StepsIndicator, Command> fieldsToCommands = CommandTree.init().getFieldsToCommands();
```
These 3 lines should be enough I believe.
```suggestion
        final Map<StepsIndicator, Command> fieldsToCommands = CommandTree.init().getFieldsToCommands();
        Command command = fieldsToCommands.get(stepsIndicator);
        userPath.add( command);
```

Y-Pazniak replied:
Sorry, don't understand your proposition. Probably, I should explain what's going on in this code.

MaximPilipeiko commented on src/main/java/by/custom/utilcalculator/domain/tree/CommandTree.java:
```diff
@@ -19,18 +21,24 @@ private CommandTree() throws UtilsborCommandTreeReadingException {
     }
 
-    public static CommandTree getInstance() {
+    public static CommandTree init() {
```
```suggestion
    public static CommandTree getInstance() {
```

Y-Pazniak replied:
Fixed

MaximPilipeiko commented on src/main/java/by/custom/utilcalculator/domain/TreeMediator.java:
```diff
@@ -0,0 +1,16 @@
+import by.custom.utilcalculator.domain.tree.CommandTree;
+
+public class TreeMediator {
+    public static boolean validateCommand(final Command requestingCommand, final UserProgress userProgress) {
```
I think this method can be moved to `CommandTree`. Additional class `TreeMediator` is redundant.

Y-Pazniak replied:
Done

MaximPilipeiko commented on src/main/java/by/custom/utilcalculator/domain/TreeMediator.java:
```diff
@@ -0,0 +1,16 @@
+        return CommandTree.init().validateCommandFromNode(command, userProgress);
+    }
+
+    public static Step getNextStep (final UserProgress userProgress){
```
I think this method can be moved to `CommandTree`. Additional class `TreeMediator` is redundant.

Y-Pazniak replied:
I can do it easily. I just tried to do less dependencies between 2 big parts of our program: tree and userProgress I decided to use Mediator to avoid direct connections between Node and UserProgress.

MaximPilipeiko replied:
`getNextStepFromNode` and `validateCommandFromNode` are already in CommandTree. For me, one more layer will not reduce dependency.

Y-Pazniak replied:
Now I see, done

MaximPilipeiko commented on src/main/java/by/custom/utilcalculator/domain/constants/Command.java:
```diff
@@ -1,70 +1,111 @@
+    //command families - not commands actually, just markers
+    AGE("age", null),
+    ENGINE_VOLUME("volume", null),
+    N1_N3_WEIGHT("n1_n3_weight", null),
```
I think you can merge   N1_N3_WEIGHT and WEIGHT into one WEIGHT. But for now, lets leave as it's.

Y-Pazniak replied:
It is better to do it now. It was just my mistake to miss this particular family. Fixed

MaximPilipeiko commented on src/main/java/by/custom/utilcalculator/domain/tree/CommandTree.java:
```diff
@@ -19,18 +21,24 @@ private CommandTree() throws UtilsborCommandTreeReadingException {
     }
 
+    public Step getNextStepFromNode(final UserProgress userProgress) {
+        System.out.println("current node: " + getNode(userProgress).getKey());
```
Please, rid `println`

Y-Pazniak replied:
I can rid it, but is it necessary for now? It does great help for me during testing the bot: I can see the current step and the next step in console: if I have an error and nodes don't work in a proper way, I can discover this error from console easily. 

It is not the first time you ask me to delete println - why is it important? Let's discuss.

MaximPilipeiko replied:
Let's discuss it on call.

MaximPilipeiko commented on src/main/java/by/custom/utilcalculator/domain/tree/Node.java:
```diff
@@ -10,12 +11,17 @@ public class Node {
     private final List<Node> children;
     private final Command key;
+    private final Step nextStep;
 
```
You can store step number. Step number is a level of the node in tree.
```suggestion
private Int number;
```

Y-Pazniak replied:
Let's discuss, how we can use int stepNumber

MaximPilipeiko commented on src/main/java/by/custom/utilcalculator/service/CalculatorExceptM1.java:
```diff
@@ -0,0 +1,155 @@
+        return CalculatorExceptM1Holder.CALCULATOR_INSTANCE;
+    }
+
+    public String calculate(final UserProgress userProgress) {
```
For me it looks like you can store the final price in the node as well.

Y-Pazniak replied:
Sounds promising, but this is a magic for me. We should discuss ways how is it possible to implement this feature

Y-Pazniak committed:
Rewriting in list has been added

Y-Pazniak committed:
Rewriting in list has been added

Y-Pazniak committed:
Rewriting in list has been added

Y-Pazniak committed:
Rewriting in list has been added

Y-Pazniak committed:
Rewriting in list has been added

Y-Pazniak committed:
Rewriting in list has been added

Y-Pazniak committed:
Steps cleaners were deleted

MaximPilipeiko commented on src/main/java/by/custom/utilcalculator/domain/UserProgress.java:
```diff
@@ -5,155 +5,158 @@
+
+        for (int i = 0; i < userPath.size(); i++) {
+            Command localCommand = userPath.get(i);
+            for (Map.Entry<StepsIndicator, Command> entry : fieldsToCommands.entrySet()) {
```
Looks like `Map<Command, StepsIndicator> commandsToFields` will be useful

Y-Pazniak replied:
_one more layer will not reduce dependency._
Probably you are right. Fixed it. 

_Looks like Map<Command, StepsIndicator> commandsToFields will be useful_
I've added a new map, but probably I've done smth wrong - I feel no big difference between before and after adding the map

MaximPilipeiko commented on src/main/java/by/custom/utilcalculator/domain/UserProgress.java:
```diff
@@ -5,155 +5,158 @@
+                        needAddCommand = false;
+                        userPath.set(i, command);
+                        lastNumber = i;
+                        break;
```
Please, check. It will be broken only from the first `for` loop but I think you would like to finish both loops.

Y-Pazniak replied:
Yes, great notice. Fixed it, thank you.

MaximPilipeiko commented on src/main/java/by/custom/utilcalculator/domain/UserProgress.java:
```diff
@@ -5,155 +5,158 @@
+        if (needAddCommand) {
+            userPath.add(command);
+        } else {
+            for (int i = lastNumber + 1; i < userPath.size(); i++) { //this code removes all cells in list after requested by user
```
I suggest to remove all elements from list instead of set null.
https://stackoverflow.com/questions/22802232/remove-all-elements-from-a-list-after-a-particular-index

Y-Pazniak replied:
That's a brilliant idea, much better than nulls. Implemented it. Thanks a lot.

MaximPilipeiko reviewed (changes requested):
Almost done

Y-Pazniak committed:
bug with wrong sequence string was fixed

fixes from PR were added

Y-Pazniak committed:
tree has been updated - new branch (self-propelled vehicles) was added

Y-Pazniak committed:
adding self-propelled vehicles - choosing particular type is in progress

Y-Pazniak committed:
adding self-propelled vehicles - choosing particular type is in progress

Y-Pazniak committed:
second step for self-propelled vehicles has been added

Y-Pazniak committed:
help request has been added

Y-Pazniak committed:
price have been added

Y-Pazniak committed:
small visual improvements have been added

Y-Pazniak committed:
text formatting and bold for price have been added

Y-Pazniak committed:
mistake with wrong loop braking was fixed

Y-Pazniak committed:
useless strings from properties file were removed

Y-Pazniak committed:
useless tree has been removed

MaximPilipeiko commented on src/main/java/by/custom/utilcalculator/controller/MessageRouter.java:
```diff
@@ -44,7 +44,11 @@ private SendMessage getCheckInputMessageAndGetAnswer(final Message message) {
             return new SendMessage(chatID, getExceptionText(e));
         }
-        return new SendMessage(chatID, answer);
+        SendMessage sendMessage = new SendMessage();
```
Using a constructor rather than setters to create an object is better. Please, read https://www.reddit.com/r/javahelp/comments/geg5ty/constructors_vs_getters_setters/. And we will discuss this question during the call.
return new SendMessage(
  chatID, answer, "HTML"
);

Y-Pazniak replied:
Done. Impossible to read an article for me, let's discuss together

MaximPilipeiko commented on src/main/java/by/custom/utilcalculator/domain/UserProgress.java:
```diff
@@ -5,155 +5,158 @@
+
+        for (int i = 0; i < userPath.size(); i++) {
+            Command localCommand = userPath.get(i);
+            for (Map.Entry<Command, StepsIndicator> entry : CommandTree.getInstance().getCommandsToFields().entrySet()) {
```
What about using `fieldsToCommands`?

Y-Pazniak replied:
I like using fieldsToCommands more.

MaximPilipeiko commented on src/main/java/by/custom/utilcalculator/domain/tree/HelperTree.java:
```diff
@@ -75,4 +212,12 @@ private static void fillParents(final Node node) {
     }
+
+    public static Map<Command, StepsIndicator> fillCommandsToFields(Map<StepsIndicator, Command> fieldsToCommands) {
+        Map<Command, StepsIndicator> commandsToFields = new HashMap<>();
```
Looks like it should be `Map<Command, List<StepsIndicator>>`. Please, look at `Command.ENGINE_VOLUME`.

Y-Pazniak replied:
I see - we did a mistake using the same key for different values. I'll fix it easily.

MaximPilipeiko commented on src/main/java/by/custom/utilcalculator/service/MessagesCreator.java:
```diff
@@ -22,22 +35,34 @@ public static MessagesCreator getInstance() {
     public String buildNextStepQuestion(final UserProgress userProgress) {
```
Maybe it's also possible to store it in tree

Y-Pazniak replied:
Can we call somehow a method from the a tree? If it is possible, it is better to replace.

MaximPilipeiko commented on src/main/java/by/custom/utilcalculator/service/UserProgressManager.java:
```diff
@@ -26,31 +26,99 @@ public static UserProgressManager getInstance() {
+        }
+
+        switch (requestingCommand) {
+            case LESS_2_TONS -> userProgress.setWeight(Weight.LESS_2_TONS);
```
You already have association between command and StepIndecator. I think you can use instead duplicate this in switch-case

Y-Pazniak replied:
🤔 it looks very promising, I'll try it

Y-Pazniak committed:
Constructor instead of creating object and filling setters has been added

Y-Pazniak committed:
fieldsToCommands instead of commandsToFields has been added.

Y-Pazniak committed:
commandsToFields Map mistake has been fixed

Y-Pazniak committed:
restored project from a great f*ck up

Y-Pazniak committed:
Confusing class SendMessageHandler.java has been removed, constructor and setter have been set in a proper way

Y-Pazniak committed:
Confusing class SendMessageHandler.java has been removed, constructor and setter have been set in a proper way

Y-Pazniak committed:
Confusing class SendMessageHandler.java has been removed, constructor and setter have been set in a proper way

Y-Pazniak committed:
Confusing class SendMessageHandler.java has been removed, constructor and setter have been set in a proper way

MaximPilipeiko commented on src/main/java/by/custom/utilcalculator/controller/Bot.java:
```diff
@@ -36,7 +36,7 @@ public String getBotUsername() {
     public String getBotToken() {
         return
-                System.getProperty("custom_chatbot_token");
+                System.getProperty("custom_chatbot_test_token");
```
It looks strange that you add `_test_` to token name.

MaximPilipeiko commented on src/main/java/by/custom/utilcalculator/domain/UserProgress.java:
```diff
@@ -5,155 +5,156 @@
-    public TransportType getTransportType(){
-        return transportType;
+    private void addUserStatusToPath(final StepsIndicator stepsIndicator) {
+        final Map<StepsIndicator, Command> fieldsToCommands = CommandTree.getInstance().getFieldsToCommands();
```
Now I think it's better to rely on tree to get step number. Let's discuss it on a call.
